pyanaconda/modules/subscription/installation.py

```diff
@@ -135,3 +137,138 @@ def run(self):
         log.debug("subscription: setting RHSM log level back to INFO")
         self._rhsm_config_proxy.Set("logging.default_log_level",
                                     get_variant(Str, "INFO"), "")
+
+
+class TransferSubscriptionTokensTask(Task):
+    """Transfer subscription tokens to the target system."""
+
+    RHSM_REPO_FILE_PATH = "/etc/yum.repos.d/redhat.repo"
+    RHSM_CONFIG_FILE_PATH = "/etc/rhsm/rhsm.conf"
+    RHSM_SYSPURPOSE_FILE_PATH = "/etc/rhsm/syspurpose/syspurpose.json"
+    RHSM_ENTITLEMENT_KEYS_PATH = "/etc/pki/entitlement"
+    RHSM_CONSUMER_KEY_PATH = "/etc/pki/consumer/key.pem"
+    RHSM_CONSUMER_CERT_PATH = "/etc/pki/consumer/cert.pem"
+
+    TARGET_REPO_FOLDER_PATH = "/etc/yum.repos.d"
+
+    def __init__(self, sysroot, transfer_subscription_tokens):
+        """Create a new task.
+
+        :param str sysroot: target system root path
+        :param bool transfer_subscription_tokens: if True attempt to transfer subscription
+                                                  tokens to target system (we always transfer
+                                                  system purpose data unconditionally)
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._transfer_subscription_tokens = transfer_subscription_tokens
+
+    @property
+    def name(self):
+        return "Transfer subscription tokens to target system"
+
+    def _copy_pem_files(self, input_folder, output_folder, not_empty=True):
+        """Copy all pem files from input_folder to output_folder.
+        Files with the pem extension are generally encryption keys and certificates.
+        If output_folder does not exist, it & any parts of its path will
+        be created.
+        :param str input_folder: input folder for the pem files
+        :param str output_folder: output folder where to copy the pem files
+        :return: False if the input directory does not exists or is empty,
+                True after all pem files have be successfully copied
+        :rtype: bool
+        """
+        # check the input folder exists
+        if not os.path.isdir(input_folder):
+            return False
+        # optionally check the input folder is not empty
+        if not_empty and not os.listdir(input_folder):
+            return False
+        # create the output folder path if it does not exist
+        if not os.path.isdir(output_folder):
+            util.mkdirChain(output_folder)
+        # transfer all the pem files in the input folder
+        for pem_file_path in glob.glob(os.path.join(input_folder, "*.pem")):
+            shutil.copy(pem_file_path, output_folder)
+        # if we got this far the pem copy operation was a success
+        return True
+
+    def _copy_file_to_path(self, file_path, target_file_path):
+        if not os.path.isfile(file_path):
+            return False
+        if not os.path.isdir(os.path.dirname(target_file_path)):
```

Comment:
You can remove this check based on our documentation `util.mkdirChain` should not fail if the target already exists.

Reply:
Cool, makes the code simpler. :)